pkg/apis/agent/v1alpha1/validations.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v1alpha1
+
+import (
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	defaultChecks = []func(*Agent) field.ErrorList{
+		checkNoUnknownFields,
+		checkNameLength,
+		checkSupportedVersion,
+		checkAtMostOneDeploymentOption,
+		checkSingleConfigSource,
+		checkSpec,
```

Comment:
Can we add one more to validate whether output is specified correctly? Agent expects exactly one output to be named `default`. To facilitate this for ECK users, they don't need to provide `outputName` if there is only one item in `elasticsearchRefs`. But they have to specify it for all outputs if there is more than one item. I think it would be good to validate user input here.